### PR TITLE
Tetrahedral remeshing - deal with c3t3 complex edges

### PR DESCRIPTION
**Pull request by janetournois**
## Summary of Changes

For a mesh generated by Mesh_3, that has complex edges that do not lie at the intersection of surface patches, automatic detection of these features is impossible from the input triangulation.
This PR introduces a new named parameter `edge_is_constrained_map` to `convert_to_triangulation_3()` that sets the edges constrained status corresponding to their `is_in_complex()` status.

This pmap can then be used straight away in `tetrahedral_isotropic_remeshing()`.

## Release Management

* Affected package(s): Tetrahedral_remeshing
* License and copyright ownership: unchanged

### CODE REVIEW

File: Tetrahedral_remeshing/doc/Tetrahedral_remeshing/Tetrahedral_remeshing.txt [hunk 1]
```diff
@@ -103,6 +103,12 @@ setting the named parameter `remesh_boundaries` to `false`.
 
 \cgalExample{Tetrahedral_remeshing/tetrahedral_remeshing_with_features.cpp }
 
+It is also possible to define the polyline features as the ones
+stored as complex edges in a `Mesh_complex_3_in_triangulation_3`
+(e.g. generated by the \ref PkgMesh3 package mesh generation algorithms).
```
Comment (albert-github) [inline Tetrahedral_remeshing/doc/Tetrahedral_remeshing/Tetrahedral_remeshing.txt:108]: Line should be:
```
(e.g. generated by the \ref PkgMesh3 "package mesh generation algorithms").
```
note the double quotes

**Comment by afabri** [comment]
Errors in the [testsuite](https://cgal.geometryfactory.com/CGAL/testsuite/#Tetrahedral_remeshing_Examples)

**Comment by sloriot** [comment]
Successfully tested in  CGAL-6.0-Ic-129

**Pull request merged**